### PR TITLE
fix(modalbox): add the small modifier to modalbox

### LINKED ISSUE #1291: Update modal examples, add support for width prop
This is a result of https://github.com/patternfly/patternfly-react/issues/1138

The PR: https://github.com/patternfly/patternfly-next/pull/1278

The modal is now full-width by default, and you can specify a `width` now if you want that could add an inline `width` to the element. 

We now have a `pf-m-sm` modifier that sets a small `max-width` on the modal (the same as the old default `max-width` with no modifier), and left the `pf-m-lg` modifier as it was.

### PR DESCRIPTION
**Pull request by boaz0**
**What**:

closes #1291 and #1138 

//cc @mcoker can you take a look to make sure I didn't miss anything. Thanks.

### CODE REVIEW

**Comment by mcoker** [comment]
thanks @boaz0, looks like the preview isn't working so I can't verify visually. though looking at the code, I'm not sure if it should be part of this PR or not, but this change was also made to enable users to pass a `width` prop to the modal component with a custom width that would be used without either `isSmall` (`.pf-m-sm`) or `isLarge` (`.pf-m-lg`) per the issue you opened https://github.com/patternfly/patternfly-react/issues/1138. Per that issue, now if you want a full width modal, you would just omit `isSmall` and `isLarge`, because the modal will now be 100% width by default.

**Comment by boaz0** [comment]
@mcoker fixed that: https://1372-pr-patternfly-react-patternfly.surge.sh/patternfly-4/components/modal

**Comment by mcoker** [comment]
Looks like there is an extra div between `div.pf-l-bullseye` and `div.pf-c-modal-box` that is causing an alignment issue. Is it possible to get rid of that or will that always be there with this component? If it will always be there, I'll need to make a small adjustment in core. 

screen shot 2019-02-13 at 10 30 28 am

**Comment by boaz0** [comment]
@mcoker unfortunately I can't remove this. This is one of the "side-effects" of [`FocusTrap`](https://github.com/patternfly/patternfly-react/blob/master/packages/patternfly-4/react-core/src/components/Modal/ModalContent.js#L53) to catch clicks outside the modal.

**Comment by boaz0** [comment]
@dlabaj I fixed that. Thanks for your review.

**Comment by redallen** [comment]
Your Travis build is likely fine, just merge master and push to get the latest CI changes.

**Comment by boaz0** [comment]
thanks @redallen - did git rebase.

**Comment by tlabaj** [comment]
@boaz0 Can you please rebase again please.  Some changes were made in Master to fix failing Build. Thanks.

**Comment by mcoker** [comment]
@boaz0 gotcha. Is there any way to add a class or otherwise style the generated element? If not we're going to need to reconsider components we currently center using the `pf-l-bullseye` layout.

**Comment by boaz0** [comment]
@mcoker looking at the documents, `<FocusTrap>` gets three props: `active`, `pasued` and `focursTrapOptions` .
None of those provide a way to set `className` or define the element that is wrapped over the modal.

So I did what you suggested and used `Bullseye` to center the modal.

**Comment by mcoker** [comment]
Gotcha, thanks. I don't think it's ideal to have to use 2 `Bullseye`'s to center something like that. We have a CSS meeting tomorrow, and I'll add this to our agenda of things to discuss.

**Comment by mcoker** [comment]
@boaz0 for now I think what you've done with 2 `Bullseye`s is fine.

is it common for a component like `<FocusTrap>` to allow for a `ClassName`? I'm curious if we can request this be added, so we could add the `.pf-l-bullseye` class to it?

**Comment by boaz0** [comment]
Opened a ticket https://github.com/davidtheclark/focus-trap-react/issues/40 for that thing.

I will update you when I will have something.

**Comment by boaz0** [comment]
Hi @mcoker I have good news. I took a look at the source code and it seems that I was wrong.
You can specify `props` like `id`, `className` or any HTML props and it will be added to the `div` element.

I am updating the code and tests and pushing it for your review.
Hope I didn't make you work too hard on this.

Thanks.

**Comment by mcoker** [comment]
@boaz0 awesome!  Good to know, I'm sure we'll use `FocusTrap` in other places and will need that, too.

File: packages/patternfly-4/react-core/src/components/Modal/Modal.js [hunk 1]
```diff
@@ -23,19 +23,25 @@ const propTypes = {
   actions: PropTypes.any,
   /** A callback for when the close button is clicked */
   onClose: PropTypes.func,
+  /** Default width of the Modal. */
```
Comment (jschuler) [inline packages/patternfly-4/react-core/src/components/Modal/Modal.js]: Should we add clarification, something like `Use either the width or the isLarge/isSmall prop`

Comment (boaz0) [reply packages/patternfly-4/react-core/src/components/Modal/Modal.js]: I will check this out.
Although if I am not wrong `isLarge` and `isSmall` are setting `maxWidth` and `minWidth` to `Modal` respectively.

Thanks.

**Comment by boaz0** [comment]
Awesome! :raised_hands:  Thank you!

**Pull request merged**